base/src/com/google/idea/blaze/base/qsync/CandidatePackageFinder.java

```diff
@@ -90,14 +90,16 @@ public ImmutableList<CandidatePackage> getCandidatePackages(
     // - at least two paths with any packages in
     // this is to offer the user some choice, and ensure we don't add paths with no packages in
     // (which would cause the query to fail later on).
+    // However, sometimes there might be a case where there's only one valid package.
+    // in such a case, we return it, but we don't add it unconditionally.
     do {
       cancellationCheck.run();
       packages = runQuery(forPath);
       if (!packages.isEmpty()) {
         candidates.add(new CandidatePackage(forPath, packages.size()));
       }
       forPath = forPath.getParent();
-    } while (candidates.size() < 2 && packages.size() < 2);
+    } while (candidates.size() < 2 && packages.size() < 2 && forPath != null);
```

Comment:
`forPath != null`

this could become an infinite loop

>Returns the parent path, or null if this path does not have a parent.

Reply:
what do you mean?

Reply:
it's a `Path`, running .getParent on it must end with a null eventually

Reply:
oh, my bad, I'm so sorry about that